inngest/src/function.rs

```diff
@@ -41,10 +42,13 @@ impl Default for FunctionOps {
     }
 }
 
+type Func<T, E> =
+    dyn Fn(Input<T>, StepTool) -> BoxFuture<'static, Result<Value, E>> + Send + Sync + 'static;
```

Comment:
changed to pass in by value instead of reference since async closures need to take ownership of them for the future to be boxed (static)